BrainFlowPlugin/Source/BrainFlowPlugin/BrainFlowPlugin.Build.cs

```diff
@@ -5,44 +5,30 @@
-            RuntimeDependencies.Add("$(TargetOutputDir)/GanglionLib32.dll", Path.Combine(ModuleDirectory, "Compiled", "Win32_dynamic", "lib", "GanglionLib32.dll"));
-            RuntimeDependencies.Add("$(TargetOutputDir)/gforce32.dll", Path.Combine(ModuleDirectory, "Compiled", "Win32_dynamic", "lib", "gforce32.dll"));
-            RuntimeDependencies.Add("$(TargetOutputDir)/gForceSDKWrapper32.dll", Path.Combine(ModuleDirectory, "Compiled", "Win32_dynamic", "lib", "gForceSDKWrapper32.dll"));
-            RuntimeDependencies.Add("$(TargetOutputDir)/neurosdk-x86.dll", Path.Combine(ModuleDirectory, "Compiled", "Win32_dynamic", "lib", "neurosdk-x86.dll"));
-            RuntimeDependencies.Add("$(TargetOutputDir)/BrainBitLib32.dll", Path.Combine(ModuleDirectory, "Compiled", "Win32_dynamic", "lib", "BrainBitLib32.dll"));
-            RuntimeDependencies.Add("$(TargetOutputDir)/eego-SDK32.dll", Path.Combine(ModuleDirectory, "Compiled", "Win32_dynamic", "lib", "eego-SDK32.dll"));
-        }
         if(Target.Platform == UnrealTargetPlatform.Win64)
         {
-            PublicLibraryPaths.Add(Path.Combine(ModuleDirectory, "Compiled", "x64_dynamic", "lib"));
             PublicIncludePaths.Add(Path.Combine(ModuleDirectory, "Compiled", "x64_dynamic", "inc"));
-            PublicAdditionalLibraries.Add("Brainflow.lib");
-            PublicAdditionalLibraries.Add("DataHandler.lib");
-            PublicAdditionalLibraries.Add("BoardController.lib");
-            PublicAdditionalLibraries.Add("MLModule.lib");
+            
+            string Folder = Path.Combine(ModuleDirectory,  "Compiled", "x64_dynamic", "lib");
+            PublicAdditionalLibraries.Add(Folder + "/Brainflow.lib");
+            
+            PublicAdditionalLibraries.Add(Folder + "/DataHandler.lib");
+            PublicAdditionalLibraries.Add(Folder + "/BoardController.lib");
+            PublicAdditionalLibraries.Add(Folder + "/MLModule.lib");
+            
             RuntimeDependencies.Add("$(TargetOutputDir)/BoardController.dll", Path.Combine(ModuleDirectory, "Compiled", "x64_dynamic", "lib", "BoardController.dll"));
             RuntimeDependencies.Add("$(TargetOutputDir)/DataHandler.dll", Path.Combine(ModuleDirectory, "Compiled", "x64_dynamic", "lib", "DataHandler.dll"));
             RuntimeDependencies.Add("$(TargetOutputDir)/GanglionLib.dll", Path.Combine(ModuleDirectory, "Compiled", "x64_dynamic", "lib", "GanglionLib.dll"));
-            RuntimeDependencies.Add("$(TargetOutputDir)/gforce64.dll", Path.Combine(ModuleDirectory, "Compiled", "x64_dynamic", "lib", "gforce64.dll"));
-            RuntimeDependencies.Add("$(TargetOutputDir)/gForceSDKWrapper.dll", Path.Combine(ModuleDirectory, "Compiled", "x64_dynamic", "lib", "gForceSDKWrapper.dll"));
+            //RuntimeDependencies.Add("$(TargetOutputDir)/gforce64.dll", Path.Combine(ModuleDirectory, "Compiled", "x64_dynamic", "lib", "gforce64.dll"));
```

Comment:
Those dlls seem to be missing (they are built by the python script but not copied over). 

Are they still relevant should I look into copying them over or should I remove them from here?

Reply:
They are not missed, they are copied if you run python script with `--oymotion` option and it seems like I forgot to add this option to readme.
I also considered an option to take dlls from this job which stored all possible libs as github artifacts https://github.com/brainflow-dev/brainflow/actions/workflows/deploy_cpp_libs.yml but ended up using submodules because it was easier for me but maybe not for users

Reply:
btw in fact there are some missing libraries inside this unreal engine build script, for example simpleble, muse_bglib, etc. Need to add them to support all possible devices. You can set all options in build.py to true to see the full list of libs

Reply:
also maybe it makes sense to write a method which will go through libs in these folders and add all of them instead hardcoding names

Reply:
> I also considered an option to take dlls from this job which stored all possible libs as github artifacts https://github.com/brainflow-dev/brainflow/actions/workflows/deploy_cpp_libs.yml but ended up using submodules because it was easier for me but maybe not for users

My suggestion, especially if you want to publish this to the maketplace sometime soon, is to push the precompiled binaries into this repo. I know it's not ideal to have binary files in git, but it really makes the publishing process to marketplace easier + you are sure if someone clones they are using a compatible version (UE SDK - C++ SDK). The job you have there seems perfect we just need to make sure it supports all the platforms you care about. I also wrote something similar here: https://github.com/outoftheboxplugins/configcat-unreal/actions/runs/5833331101

It's still very important to give the users some instructions about compiling locally because they might need to change the C++ SDK.

> also maybe it makes sense to write a method which will go through files in lib folder after compilation and add all of them ro runtimedependencies. It will look better than hardcoding names, especially including the fact that there are different build options

Yes, it makes total sense. I actually did something similar a few weeks ago for this plugin: https://github.com/outoftheboxplugins/configcat-unreal/blob/main/Source/ThirdParty/ConfigCatCppSDK/ConfigCatCppSdk.Build.cs

I wanted to keep the changes of the PR to a minimum, but I am happy to implement it here for this PR or next ones.

> btw in fact there are some missing libraries inside this unreal engine build script, for example simpleble, muse_bglib, etc. Need to add them to support all possible devices. You can set all options in build.py to true to see the full list of libs

I think if we deliver pre-built binaries we can just include all in the repo and we can adjust the `Build.cs` accordingly.

Reply:
I think we can setup jobs to compile everything and push to marketplace wo pushing binaries to git, but can push binaries also. The main requirement - it should be simple to update and dont take too much time to maintain, so probably smth like cron job with personal access token will be ok.

Lets fix this ugly hardcode for dyn libs and iterate over files in libs dir as a part of this PR

Reply:
Sounds good, I've adjust the DLLs and the libs discovery.

Since this is a bigger change, I would love to know if you can suggest some tests to confirm everything is working well. (I've tried building some bits from https://brainflow.readthedocs.io/en/stable/Examples.html#id2). Or if you know someone who's using this with Unreal, I would be happy to get in touch with them and confirm it works for them.

<img width="1553" alt="Screenshot 2023-09-11 at 08 30 27" src="https://github.com/brainflow-dev/BrainFlowUnrealPlugin/assets/21221169/88c1cddc-32a4-4bbb-9206-7f90e0c36f92">

Reply:
this test is ok you can use it with synthetic board and see if you can read some data

Reply:
did you manage to get some data using this sample?

Reply:
Yes. Thanks again for all the help both here and on slack. I was able to get the sample data both for at editortime (Play in Editor) and runtime (packaged game).

I've modified the file as discussed, the brainflow.lib is statically linked in the .exe and the 3 DLL files (board controller, ml module, data handler) are dynamically linked. All other DLL are copied over to the final build so they can loaded if needed.